### PR TITLE
Fonts

### PR DESCRIPTION
**Pull request by NurkaAmre**
## Context

[(https://ywgc.atlassian.net/jira/software/projects/YWGC/boards/1?selectedIssue=YWGC-14)]

## Work done
 Separate fonts-related changes into a dedicated pull request

### CODE REVIEW

**Review by novellac — CHANGES_REQUESTED** [review]
Request: It looks like you've got some of the work from your other PR accidentally carried over here. Let's keep it to one PR one task (in this case, replacing the fonts). Remove the unrelated changes and ping me again, because it is looking pretty good!

File: src/pages/Layout.jsx
Comment (novellac) [inline src/pages/Layout.jsx:1]: issue: These changes aren't related, so let's not include them in the PR. This will help ensure we have fewer merge conflicts and a clearer code history.

File: src/pages/Home.jsx
Comment (novellac) [inline src/pages/Home.jsx:1]: issue: These changes aren't related, so let's not include them in the PR. This will help ensure we have fewer merge conflicts and a clearer code history.
observation(nb): It looks like you might have old code - have you updated your branch with the latest changes from `main`? Reach out to me or @alliequintano if you need help doing that.

File: public/index.html [hunk 1]
```diff
@@ -20,6 +15,8 @@
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link href="https://fonts.googleapis.com/css2?family=Fjalla+One&family=Merriweather+Sans:wght@300&display=swap"
+      rel="stylesheet">
```
Comment (novellac) [inline public/index.html:19]: issue: Double-check the fonts you're pulling here with what's in Figma - I think the design requires may require more weights and/or styles. Check in with @ajkarczewski if you need help finding the page in Figma with the design elements on it.

File: package.json
Comment (novellac) [inline package.json:1]: issue: These changes aren't related to the fonts PR, so they should be done in their own PR. Since this is just reordering the file, I wouldn't normally care too much, but since we've had some rough merge conflicts, I want to keep our PRs as small and narrowly-scoped as possible.

File: tailwind.config.js [hunk 2]
```diff
@@ -5,7 +5,13 @@ module.exports = {
   "./src/**/*.{js,jsx,ts,tsx}",
   ],
   theme: {
-  extend: {},
+    extend: {
+      fontFamily: {
+        header: ['Fjalla One', 'sans-serif'],
+        body: ['Merriweather Sans', 'sans-serif'],
+        button: ['Merriweather Sans', 'sans-serif'],
```
Comment (novellac) [inline tailwind.config.js:12]:
```suggestion
        'sans': ['Merriweather Sans', ...defaultTheme.fontFamily.sans]
```
nitpick: It's not wrong to do this, since it will allow us to have a `font-body` class, but it's unnecessary. Instead, I'd suggest overriding the default `sans` (which TW applies to our HTML by default).
suggestion(nb): Read more about setting [default fonts in TW](https://tailwindcss.com/docs/font-family#customizing-the-default-font).

File: public/index.html [hunk 3]
```diff
@@ -19,7 +15,7 @@
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <link href="https://fonts.googleapis.com/css2?family=Fjalla+One&family=Merriweather+Sans:wght@300&display=swap"
+    <link href="https://fonts.googleapis.com/css2?family=Fjalla+One&family=Merriweather+Sans:wght@300;400;700&display=swap"
```
Comment (novellac) [inline public/index.html:18]: question(nb): I'm not quite sure how it works - does importing Fjalla One without any font weights mean we get all the weights? Or what?

File: src/components/BaseHeading.jsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ const BaseHeading = ({ level, text }) => {
   const HeadingTag = `h${level}`
 
   return (
-    <HeadingTag className="text-4xl text-center md:text-6xl">
+    <HeadingTag className="text-4xl text-center font-header md:text-6xl">
```
Comment (novellac) [inline src/components/BaseHeading.jsx:8]: Praise(nb): Good use of the `font-header` class you created! Now wherever we use `BaseHeading` we will get this font, and in addition we can use it in other places where it's appropriate.

File: src/pages/Home.jsx [hunk 5]
```diff
@@ -6,12 +6,12 @@ function Home () {
     <div className="p-6 bg-white bg-opacity-60 bg-[url('/public/lawncare-bg.jpeg')] bg-no-repeat bg-cover">
       <div className="flex flex-col items-center justify-center p-6 mt-24 bg-white rounded-2xl">
         <BaseHeading level={1} text="Yorkshire Woods Ground Crew" />
-        <p className="mt-4 text-xl font-body">
+        <p className="mt-4 font-sans text-xl">
```
Comment (novellac) [inline src/pages/Home.jsx:9]: issue: Since we customised the `sans` fontFamily in `tailwind.config.js`, and since we know that [Tailwind applies this to everything in the `html` element](https://tailwindcss.com/docs/font-family#customizing-the-default-font), we should not need to add `font-sans` class to anything - we should now get it for free!

File: src/pages/Home.jsx [hunk 6]
```diff
@@ -6,12 +6,12 @@ function Home () {
...
           Yorkshire Woods Grounds Crew is a Detroit local lawn care and landscaping business. We specialize in lawn-mowing and clean-outs. We handle residential, business, and government properties of all sizes. We&apos;re available for one-time and ongoing engagements.
         </p>
         <a
           href='tel:+123456789'
-          className="items-center w-full p-2 my-20 text-lg font-bold text-center text-white bg-green-700 rounded-lg font-body md:mt-16 hover:bg-green-600"
+          className="items-center w-full p-2 my-20 font-sans text-lg text-center text-white bg-green-700 rounded-lg font-body md:mt-16 hover:bg-green-600"
```
Comment (novellac) [inline src/pages/Home.jsx:14]: issue: Same as above - we can remove any `font-sans` since we get it for free.

**Review by novellac — CHANGES_REQUESTED** [review]
Almost there! 
bug: I noticed that the new Merriweather font is not being applied, even though it has been added to the `sans` fontFamily. Hint: Have a look at `index.css` - you'll notice we are overriding the fontFamily that TW sets on the `html` element with our own CSS declaration on the `body` element. I think we can safely remove the whole body CSS (lines 5-12).

**Comment by novellac** [comment]
Nitpick: Please provide manual testing instructions. I realise that they might be a bit vague for this PR, but that's ok! As frontend developers we may not have dedicated QA, and we should know how to instruct others to test our PRs. Something like "Verify that all the header text is Fjalla One and all the other text is Merriweather Sans" would work.

File: public/index.html (hunk 3 above)
Comment (NurkaAmre) [reply public/index.html:18]: yes

**Review by novellac — APPROVED** [review]
Exciting! There's nothing blocking that I'd change so let's go ahead and smash that merge button! 🎉 ❤️
